src/anonymousUserTracking/anonymousUserEventManager.ts

```diff
@@ -0,0 +1,367 @@
+        delete event.eventType;
+        switch (eventType) {
+          case TRACK_EVENT: {
+            this.track(event);
+            break;
+          }
+          case TRACK_PURCHASE: {
+            this.trackPurchase(event);
+            break;
+          }
+          case TRACK_UPDATE_CART: {
+            this.updateCart(event);
+            break;
+          }
+          case UPDATE_USER: {
+            this.updateUser({ dataFields: event });
+            break;
+          }
+          default:
+            break;
+        }
+
+        localStorage.removeItem(SHARED_PREFS_ANON_SESSIONS);
+        localStorage.removeItem(SHARED_PREFS_EVENT_LIST_KEY);
+        localStorage.removeItem(SHARED_PREF_MATCHED_CRITERIAS);
+      });
+    }
+  }
+
+  private async storeEventListToLocalStorage(
+    newDataObject: Record<any, any>,
+    shouldOverWrite: boolean
+  ) {
+    const strTrackEventList = localStorage.getItem(SHARED_PREFS_EVENT_LIST_KEY);
+    let previousDataArray = [];
+
+    if (strTrackEventList) {
+      previousDataArray = JSON.parse(strTrackEventList);
+    }
+
+    if (shouldOverWrite) {
+      const trackingType = newDataObject[SHARED_PREFS_EVENT_TYPE];
+      const indexToUpdate = previousDataArray.findIndex(
+        (obj: any) => obj[SHARED_PREFS_EVENT_TYPE] === trackingType
+      );
+      if (indexToUpdate !== -1) {
+        const dataToUpdate = previousDataArray[indexToUpdate];
+
+        previousDataArray[indexToUpdate] = {
+          ...dataToUpdate,
+          ...newDataObject
+        };
+      } else {
+        previousDataArray.push(newDataObject);
+      }
+    } else {
+      previousDataArray.push(newDataObject);
+    }
+
+    localStorage.setItem(
+      SHARED_PREFS_EVENT_LIST_KEY,
+      JSON.stringify(previousDataArray)
+    );
+    const criteriaId = await this.checkCriteriaCompletion();
+    if (criteriaId !== null) {
+      this.createKnownUser(criteriaId);
+    }
+  }
+
+  private getCurrentTime = () => {
+    return new Date().getTime();
+  };
+
+  private getWebPushOptnIn(): string {
+    const notificationManager = window.Notification;
+    if (notificationManager && notificationManager.permission === 'granted') {
+      return window.location.hostname;
+    } else {
+      return '';
+    }
+  }
+
+  track = (payload: InAppTrackRequestParams) => {
+    return baseIterableRequest<IterableResponse>({
+      method: 'POST',
+      url: ENDPOINTS.event_track.route,
+      data: payload,
+      validation: {
+        data: trackSchema
+      }
+    });
+  };
+
+  updateCart = (payload: UpdateCartRequestParams) => {
+    return baseIterableRequest<IterableResponse>({
+      method: 'POST',
+      url: ENDPOINTS.commerce_update_cart.route,
+      data: {
+        ...payload,
+        user: {
+          ...payload.user,
+          preferUserId: true
+        }
+      },
+      validation: {
+        data: updateCartSchema
+      }
+    });
+  };
+
+  trackPurchase = (payload: TrackPurchaseRequestParams) => {
```

Comment:
There's really no way to reuse the other methods without a circular dependency?

Reply:
I don't think so unless we make baseIterable request-related code common like below and use it either on AnonymousUserEventManager or everywhere

```
return baseIterableRequest<IterableResponse>({
      method: 'POST',
      url: ENDPOINTS.commerce_update_cart.route,
      data,
      validation: validationSchema
    });
```